.github/workflows/pr-golangci-lint.yaml

```diff
@@ -0,0 +1,35 @@
+name: PR golangci-lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  golangci:
```

Comment:
I'm thinking we probably don't need a separate job for this if it already runs as part of the `verify` job and the codecov job.

Reply:
We dont seem to be running linter as part of `verify` job
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f0ec78f028530597f8b5f78894c085ad0a2bfd0a/Makefile#L258-L259

Reply:
I dont know how Codecov is running the linter, @willie-yao Do you know where the linter is configured? 
I do see the linter getting invoked in the output of `coverage` but cant trace its origin.

Reply:
codecov runs `make test-cover`, `test-cover` depends on `test`, `test` depends on `lint`

And I guess it's the `test` job instead of the `verify` job that also runs golangci-lint through basically the same path.

Reply:
`test: generate lint go-test-race` missed that lint there. Thanks!

Reply:
I'd vote to have this as a separate GitHub workflow job
1. Consistent with CAPI
2. Running as part of codecov, where we are trying to see the test coverage, does not align with seeing any linter errors.